mne/beamformer/tests/test_lcmv.py

```diff
@@ -746,56 +747,84 @@ def test_lcmv_reg_proj(proj, weight_norm):
-    (0.05, 'max-power', 'unit-noise-gain', False, 17, 19),
-])
+@pytest.mark.parametrize(
+    'reg, pick_ori, weight_norm, use_cov, depth, lower, upper', [
+        (0.05, 'vector', 'unit-noise-gain', True, None, 36, 39),
+        (0.05, 'vector', 'unit-noise-gain', False, None, 23, 25),
+        (0.05, 'vector', 'nai', True, None, 36, 39),
+        (0.05, 'vector', None, True, None, 12, 14),
+        (0.05, 'vector', None, True, 0.8, 39, 43),
+        # (0.00, 'vector', 'unit-noise-gain', True, None, 43, 46),  # complex
+        (0.05, 'max-power', 'unit-noise-gain', True, None, 20, 24),
+        # (0., 'max-power', 'unit-noise-gain', True, None, 37, 40),  # complex
+        (0.05, 'max-power', 'unit-noise-gain', False, None, 17, 19),
+        (0.05, 'max-power', 'nai', True, None, 20, 24),
+        (0.05, 'max-power', None, True, None, 7, 9),
+        (0.05, 'max-power', None, True, 0.8, 16, 19),
+        (0.05, None, None, True, 0.8, 40, 42),
+    ])
 def test_localization_bias_free(bias_params_free, reg, pick_ori, weight_norm,
-                                use_cov, lower, upper):
+                                use_cov, depth, lower, upper):
     """Test localization bias for free-orientation LCMV."""
     evoked, fwd, noise_cov, data_cov, want = bias_params_free
     if not use_cov:
         evoked.pick_types('grad')
         noise_cov = None
     loc = apply_lcmv(evoked, make_lcmv(evoked.info, fwd, data_cov, reg,
                                        noise_cov, pick_ori=pick_ori,
-                                       weight_norm=weight_norm)).data
+                                       weight_norm=weight_norm,
+                                       depth=depth)).data
     loc = np.linalg.norm(loc, axis=1) if pick_ori == 'vector' else np.abs(loc)
     # Compute the percentage of sources for which there is no loc bias:
     perc = (want == np.argmax(loc, axis=0)).mean() * 100
     assert lower <= perc <= upper
 
 
+@pytest.mark.parametrize('weight_norm', ('nai', 'unit-noise-gain'))
+@pytest.mark.parametrize('pick_ori', ('vector', 'max-power', None))
+def test_depth_does_not_matter(bias_params_free, weight_norm, pick_ori):
+    """Test that depth weighting does not matter for normalized filters."""
+    evoked, fwd, noise_cov, data_cov, _ = bias_params_free
+    data = apply_lcmv(evoked, make_lcmv(
+        evoked.info, fwd, data_cov, 0.05, noise_cov, pick_ori=pick_ori,
+        weight_norm=weight_norm, depth=0.)).data
+    data_depth = apply_lcmv(evoked, make_lcmv(
+        evoked.info, fwd, data_cov, 0.05, noise_cov, pick_ori=pick_ori,
+        weight_norm=weight_norm, depth=1.)).data
+    assert data.shape == data_depth.shape
+    for d1, d2 in zip(data, data_depth):
+        # Sign flips can change when nearly orthogonal to the normal direction
+        d2 *= np.sign(np.dot(d1.ravel(), d2.ravel()))
+        atol = np.linalg.norm(d1) * 1e-7
+        assert_allclose(d1, d2, atol=atol)
```

Comment:
@britta-wstnr this verifies that `depth=0.` is the same as `depth=1.` for both `'nai'` and `'unit-noise-gain'` modes